.gitlab/quartz.yml

```diff
@@ -15,131 +15,100 @@
-  extends: [.build_metis, .with_gcc_6_1_0, .on_quartz]
-  variables:
-    METIS: metis5
-  allow_failure: true
+# Spack helped builds
+# Generic quartz build job, extending build script
+.build_and_test_on_quartz:
+  extends: [.build_toss_3_x86_64_ib_script, .on_quartz]
+  stage: q_build_and_test
 
 # Build MFEM
-build_mfem_debug_ser_gcc_4_9_3_quartz:
-  extends: [.build_mfem, .with_gcc_4_9_3, .on_quartz]
+debug_ser_gcc_4_9_3:
   variables:
-    MFEM_DEBUG: "YES"
-    MFEM_USE_MPI: "NO"
+    SPEC: "%gcc@4.9.3 +debug~mpi"
+  extends: .build_and_test_on_quartz
 
-build_mfem_debug_ser_gcc_6_1_0_quartz:
-  extends: [.build_mfem, .with_gcc_6_1_0, .on_quartz]
+debug_ser_gcc_6_1_0:
   variables:
-    MFEM_DEBUG: "YES"
-    MFEM_USE_MPI: "NO"
+    SPEC: "%gcc@6.1.0 +debug~mpi"
+  extends: .build_and_test_on_quartz
 
-build_mfem_debug_par_gcc_6_1_0_quartz:
-  extends: [.build_mfem, .with_gcc_6_1_0, .on_quartz]
+debug_par_gcc_6_1_0:
   variables:
-    MFEM_DEBUG: "YES"
-    MFEM_USE_MPI: "YES"
-  needs: [build_hypre_gcc_6_1_0_quartz, build_parmetis_gcc_6_1_0_quartz]
+    SPEC: "%gcc@6.1.0 +debug+mpi"
+  extends: .build_and_test_on_quartz
 
-build_mfem_opt_ser_gcc_6_1_0_quartz:
-  extends: [.build_mfem, .with_gcc_6_1_0, .on_quartz]
+opt_ser_gcc_6_1_0:
   variables:
-    MFEM_DEBUG: "NO"
-    MFEM_USE_MPI: "NO"
+    SPEC: "%gcc@6.1.0 ~mpi"
+  extends: .build_and_test_on_quartz
 
-build_mfem_opt_par_gcc_6_1_0_quartz:
-  extends: [.build_mfem, .with_gcc_6_1_0, .on_quartz]
+opt_par_gcc_6_1_0:
   variables:
-    MFEM_DEBUG: "NO"
-    MFEM_USE_MPI: "YES"
-  needs: [build_hypre_gcc_6_1_0_quartz, build_parmetis_gcc_6_1_0_quartz]
+    SPEC: "%gcc@6.1.0"
+  extends: .build_and_test_on_quartz
 
-### Tests
-
-# Sanity check
-sanitycheck_mfem_debug_ser_gcc_4_9_3_quartz:
-  extends: [.sanitycheck_mfem, .with_gcc_4_9_3, .on_quartz]
+opt_par_gcc_6_1_0_sundials:
   variables:
-    MFEM_DEBUG: "YES"
-    MFEM_USE_MPI: "NO"
-  needs: [build_mfem_debug_ser_gcc_4_9_3_quartz, allocate_quartz]
+    SPEC: "%gcc@6.1.0 +sundials"
+  extends: .build_and_test_on_quartz
 
-sanitycheck_mfem_debug_ser_gcc_6_1_0_quartz:
-  extends: [.sanitycheck_mfem, .with_gcc_6_1_0, .on_quartz]
+opt_par_gcc_6_1_0_petsc:
   variables:
-    MFEM_DEBUG: "YES"
-    MFEM_USE_MPI: "NO"
-  needs: [build_mfem_debug_ser_gcc_6_1_0_quartz, allocate_quartz]
+    SPEC: "%gcc@6.1.0 +petsc ^petsc+mumps"
+  extends: .build_and_test_on_quartz
 
-sanitycheck_opt_ser_mfem_gcc_6_1_0_quartz:
-  extends: [.sanitycheck_mfem, .with_gcc_6_1_0, .on_quartz]
+opt_par_gcc_6_1_0_pumi:
   variables:
-    MFEM_DEBUG: "NO"
-    MFEM_USE_MPI: "NO"
-  needs: [build_mfem_opt_ser_gcc_6_1_0_quartz, allocate_quartz]
+    SPEC: "%gcc@6.1.0 +pumi"
+  extends: .build_and_test_on_quartz
 
-sanitycheck_opt_par_mfem_gcc_6_1_0_quartz:
-  extends: [.sanitycheck_mfem, .with_gcc_6_1_0, .on_quartz]
-  variables:
-    MFEM_DEBUG: "NO"
-    MFEM_USE_MPI: "YES"
-  needs: [build_mfem_opt_par_gcc_6_1_0_quartz, allocate_quartz]
+#serac_gcc_8_3_1:
```

Comment:
Is this commented code being kept here for a reason?

Reply:
I agree. I assume this is to eventually test our special flavor of builds, but I think keeping the file clean for now is preferred.